plasmapy/formulary/relativity.py

```diff
@@ -150,3 +156,354 @@ def relativistic_energy(m: u.kg, v: u.m / u.s) -> u.Joule:
+        of the relativistic body. If it is a |Quantity|, then it must
+        have units of mass.
+
+    V : |Quantity|, optional
+        The velocity of the relativistic body in units convertible to
+        m/s. The absolute magnitude of ``V`` cannot be greater than
+        :math:`c`\ .
+
+    momentum : |Quantity|, optional
+        The momentum of the relativistic body in units convertible to
+        kg·m/s.
+
+    total_energy : |Quantity|, optional, |keyword-only|
+       The sum of the mass energy and the kinetic energy in units
+       convertible to joules. Must be non-negative.
+
+    kinetic_energy : |Quantity|, optional, |keyword-only|
+       The kinetic energy of the relativistic body in units convertible
+       to joules. Must be non-negative.
+
+    v_over_c : real number or |Quantity|, optional, |keyword-only|
+       The ratio of the velocity to the speed of light. Must have an
+       absolute magnitude :math:`≤ 1`\ .
+
+    lorentz_factor : real number or |Quantity|, optional, |keyword-only|
+       The Lorentz factor of the relativistic body. Must be
+       :math:`≥ 1`\ .
+
+    Z : integer, optional, |keyword-only|
+        The charge number associated with ``particle``.
+
+    mass_numb : integer, optional, |keyword-only|
+        The mass number associated with ``particle``.
+
+    dtype : |DTypeLike|, optional, |keyword-only|, default: `numpy.longdouble`
+        The `numpy` data type to use to store the inputs.
+
+    Notes
+    -----
+    At most one of ``V``, ``momentum``, ``total_energy``,
+    ``kinetic_energy``, ``v_over_c``, and ``lorentz_factor`` must be
+    provided.
+
+    .. caution::
+
+       For ultra-high-energy cosmic rays (UHECRs), the velocity may be
+       within roundoff error of :math:`c` and :math:`\frac{V}{c}` may be
+       within roundoff error of 1.
+
+    Examples
+    --------
+    >>> import astropy.units as u
+    >>> relativistic_proton = RelativisticBody("p+", total_energy = 1 * u.GeV)
+    >>> relativistic_proton.particle
+    Particle("p+")
+    >>> relativistic_proton.velocity
+    <Quantity 1.03697...e+08 m / s>
+    >>> relativistic_proton.v_over_c
+    0.3458980898746...
+    >>> relativistic_proton.lorentz_factor
+    1.0657889247888...
+    >>> relativistic_proton.mass_energy.to("GeV")
+    <Quantity 0.93827... GeV>
+    >>> relativistic_proton.total_energy.to("GeV")
+    <Quantity 1. GeV>
+    >>> relativistic_proton.mass
+    <Quantity 1.67262...e-27 kg>
+
+    |RelativisticBody| also works with multiple particles and/or
+    velocities.
+
+    >>> particles = ["p+", "e-"]
+    >>> velocities = [2e5, 2e8] * u.m / u.s
+    >>> relativistic_particles = RelativisticBody(particles, velocities)
+    >>> relativistic_particles.momentum
+    <Quantity [3.345244...e-22, 2.445659...e-22] kg m / s>
+    """
+
+    def _get_speed_like_input(
+        self,
+        velocity_like_arguments: Dict[str, Union[u.Quantity, Real]],
+    ):
```

Comment:
```suggestion
    @staticmethod
    def _get_speed_like_input(
        velocity_like_arguments: Dict[str, Union[u.Quantity, Real]],
    ):
```

I like doing these whenever `self` is not used, otherwise LSP complains at me while I'm writing these.